pkg/kube/events.go

```diff
@@ -54,12 +54,19 @@ func NewSkupperEventRecorder(namespace string, cli kubernetes.Interface) *Skuppe
 	kubeEventRecorder := eventBroadcaster.NewRecorder(
 		scheme.Scheme,
 		v1.EventSource{
-			Component: types.ControllerServiceName})
-	service, _ := cli.CoreV1().Services(namespace).Get(context.TODO(), types.ControllerServiceName, metav1.GetOptions{})
+			Component: types.ControllerDeploymentName})
+	deployment, _ := cli.AppsV1().Deployments(namespace).Get(context.TODO(), types.ControllerDeploymentName, metav1.GetOptions{})
```

Comment:
This assumes that the service-controller deployment is always present in the namespace for which the event recorder is being used. That is true now, but might not always be so.

One option could be to have the source provided as an argument to this function, and then to have the service-controller (which I think is at present the only user of the function) provide an object ref to its own deployment.

Reply:
Makes sense! updated in 6b750cc0e2b00989847c9bf47d4e4a207903dc07